### PR TITLE
DBZ-872 Honor filters in ALTER TABLE parsing

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-872

### CODE REVIEW

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/listener/AlterTableParserListener.java [hunk 1]
```diff
@@ -122,6 +125,7 @@ public void exitColumnDefinition(MySqlParser.ColumnDefinitionContext ctx) {
                 else {
                     // all columns parsed
                     // reset global variables for next parsed statement
+                    columnEditors.forEach(columnEditor -> tableEditor.addColumn(columnEditor.create()));
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/listener/AlterTableParserListener.java]: What's the purpose of this, given it's set to `null` on the next line?

Comment (jpechane) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/listener/AlterTableParserListener.java]: @gunnarmorling The editors created by `ALTER TABLE` are added to the table.

Comment (gunnarmorling) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/listener/AlterTableParserListener.java]: Hum, I'm still confused. What bug is fixed exactly by this? +1 for extracting it and making it more clear. I've pushed one more commit for the change discussed on the chat; can you rework the PR to split up that first commit here? In general let's keep unrelated things in separate commits for the sake of comprehensibility :)

**Comment by jpechane** [comment]
@gunnarmorling Commit split

**Comment by gunnarmorling** [comment]
@jpechane There are two test failures; any idea what's going on? Or is it actually related to the other recent work?

**Comment by jpechane** [comment]
@gunnarmorling Fixed, somehow the valu converters needed for default value parsing were removed from the parser constructor call.

**Comment by gunnarmorling** [comment]
Rebased and applied. Thanks, @jpechane!

**Pull request merged**